### DISCUSSION TITLE
One Workflow for Me, Another for Others

### DISCUSSION THREAD

**Discussion by snayff** [community/community#26369] [Actions]
<p>With  the help of this fine community I previously set up a workflow that lints my code and pushes it back to the pull request. I was told this would not work when coming from a fork as it used <code>ref: ${{ github.event.pull_request.head.ref }} </code>.</p>
<p>So, I now have a pull request from a fork and as expected the workflow fails with “branch not found”. I’d like to run the workflow only if it comes from me and isnt from a fork. I’ve set up a simpler workflow to run for others and I need this to run when it isnt me.</p>
<p>I tried the following but got an error of “Unexpected symbol: ‘“Snayff”’.” Even if that didnt error out I wonder if I am looking at the wrong property.</p>
<pre><code class="lang-auto">name: NQP - PR Review Snayff

on:
  pull_request:
    types: [opened, edited, synchronize]

jobs:
  job1:
    if: github.event.pusher.name == "Snayff"
</code></pre>
<p>Any help would be appreciated.</p>
<p>Snayff</p>

**Comment by snayff** [comment]
<p>Any ideas?  Or is this not possible?</p>

**Comment by Simran-B** [comment, marked as answer]
<p>You need to use single quotes in expressions: <code>'Snayff'</code></p>
<p>You should also take a look here:</p>
<ul>
<li><a href="https://github.community/t/is-there-a-way-to-tell-if-a-pr-is-from-a-forked-repository/134186" class="inline-onebox">Is there a way to tell if a PR is from a forked repository?</a></li>
<li><a href="https://github.community/t/have-github-action-only-run-on-master-repo-and-not-on-forks/140840/18" class="inline-onebox">Have github action only run on master repo, and not on forks?</a></li>
<li><a href="https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662/10" class="inline-onebox">How to trigger an action on push or pull request but not both?</a></li>
</ul>

**Comment by snayff** [comment]
<p>You star, thank you - that was exactly it. I checked the following at job level.</p>
<pre><code class="lang-auto">if: ${{ github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name }}
</code></pre>
